core/math/math_funcs.h

```diff
@@ -46,7 +46,8 @@ class Math {
 public:
 	Math() {} // useless to instance
 
-	static const uint64_t RANDOM_MAX = 0xFFFFFFFF;
+	// Not using 'RANDOM_MAX' to avoid conflict with system headers on some OSes (at least NetBSD).
+	static const uint64_t RANDOM_UINT32_MAX = 0xFFFFFFFF;
```

Comment:
> `UINT32`

> `uint64_t`

Hmm...

Reply:
It's the max value of a 32-bit unsigned integer, stored in `uint64_t`. Is there a problem?

Reply:
Is there a particular reason this shouldn't be either `uint32_t` or `0xFFFFFFFFFFFFFFFF`?

Anyway, the name seems a bit strange and some may mistake it for Hungarian notation. Maybe `RANDOM_32BIT_MAX` would be clearer?

Reply:
> Is there a particular reason this shouldn't be either `uint32_t` or `0xFFFFFFFFFFFFFFFF`?

No idea.

> Anyway, the name seems a bit strange and some may mistake it for Hungarian notation. Maybe `RANDOM_32BIT_MAX` would be clearer?

Sounds fine.